pkg/crane/get.go

```diff
@@ -35,11 +40,20 @@ func getImage(r string, opt ...Option) (v1.Image, name.Reference, error) {
 	return img, ref, nil
 }
 
-func getManifest(r string, opt ...Option) (*remote.Descriptor, error) {
+func get(r string, opt ...Option) (DigestManifester, error) {
 	o := makeOptions(opt...)
 	ref, err := name.ParseReference(r, o.name...)
 	if err != nil {
 		return nil, fmt.Errorf("parsing reference %q: %v", r, err)
 	}
-	return remote.Get(ref, o.remote...)
+	desc, err := remote.Get(ref, o.remote...)
+	if err != nil {
+		return nil, err
+	}
+
+	if o.platform == nil {
+		return desc.ImageIndex()
```

Comment:
This will fail for anything that's not an index (schema1 or image) -- I'd probably just inline the logic from `getManifest` into `Manifest` and `Digest` -- the awkwardness of trying to fit this into the type system makes it tough.

This would be a lot easier if I hadn't (stupidly) embedded `v1.Descriptor` into `remote.Descriptor`, because we have a name collision between the `Digest` in the struct in the `Digest()` method 😞